library/std/src/io/stdio.rs

```diff
@@ -873,6 +875,52 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Reads a [`String`] from [standard input](Stdin). The trailing
+/// newline is stripped. Gives an error on EOF (end of file).
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(io_input)]
+/// use std::io;
+///
+/// fn main() -> io::Result<()> {
+///     print!("Enter name: ");
+///
+///     let name: String = io::read_line()?;
+///
+///     println!("Your name is {}!", name);
+///
+///     Ok(())
+/// }
+/// ```
+#[unstable(
+    feature = "io_input",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn read_line() -> Result<String> {
+    stdout().flush()?; // print!() does not flush :(
+    let mut input = String::new();
+    match stdin().read_line(&mut input)? {
+        0 => Err(Error::new(ErrorKind::UnexpectedEof, "input reached eof unexpectedly")),
+        _ => {
+            if Some('\n') == input.chars().next_back() {
```

Comment:
You could use `.ends_with()` to simplify this:

```rust
    if input.ends_with('\n') {
        input.pop();
        if cfg!(windows) && input.ends_with('\r') {
            input.pop();
        }
    }
```

Looks like that's also a lot easier to optimize for the compiler: https://godbolt.org/z/1Gc8nM

Reply:
Why not just pop without the checks?

Reply:
> Why not just pop without the checks?

@pickfire See https://github.com/rust-lang/rust/pull/75435#discussion_r495845209